### ISSUE TITLE
RectifiedAdam is not supporting scheduling for weight decays, while AdamW does

### ISSUE THREAD

**Issue by leandro-gracia-gil** [tensorflow/addons#1908]
The current implementation of RectifiedAdam implements weight decay following the [RAdam paper](https://arxiv.org/pdf/1908.03265v1.pdf), which adopts it from the [AdamW paper](https://arxiv.org/pdf/1711.05101.pdf). However, AdamW actually supports scheduling weight decay as part of its AdamWR variant, and in fact the paper defines a normalized weight decay in its appendix B.1 to handle these situations.

The current [AdamW implementation](https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/AdamW) accepts Callable arguments for both learning_rate and weight_decay, while the [RectifiedAdam implementation](https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/RectifiedAdam) only does so for the learning_rate. While it might be possible to get a decaying weight decay by providing a tensor implementing it, this hurts serialization since the state of the decay is no longer saved with the optimizer.

Looking at the current code it seems that weight decay would need to be decayed in a way similar as done [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L851).

**Relevant information**
- Are you willing to contribute it (yes/no): willing yes but most likely won't have the time in a while, I'd appreciate if somebody else can take a quick look.
- Are you willing to maintain it going forward? (yes/no): no
- Is there a relevant academic paper? (if so, where): listed above in the description.
- Is there already an implementation in another framework? (if so, where): there should be one in AdamW in this same framework.
- Was it part of tf.contrib? (if so, where): for AdamW it was tf.contrib.opt.AdamWOptimizer. There was no RectifiedAdam AFAIK.

**Which API type would this fall under (layer, metric, optimizer, etc.)**
Optimizer

**Who will benefit with this feature?**
Anyone using AdamWR who wants to gets the benefits of RectifiedAdam and its derived optimizers (like Ranger).

**Any other info.**
None.

**Comment by leandro-gracia-gil** [comment]
I tried using schedulers for weight decay with the current AdamW implementation and I'm getting an exception despite being supposedly supported (according to the argument typing).

Maybe this should be reported as a separate issue? It looks like both issues could be fixed together, and at least in the AdamW case this is already supposed to work.
